### PR TITLE
doc about taint shutdown condition (#58635)

### PR DESCRIPTION
**Pull request by mujtabaahmed1991**

### CODE REVIEW

File: contributors/design-proposals/scheduling/taint-node-by-condition.md [hunk 1]
```diff
@@ -15,26 +15,32 @@ Currently (1.8 and before), the conditions of nodes are updated by the kubelet a
...
+|NetworkUnavailable    |True                | NoSchedule   | node.kubernetes.io/network-unavailable         |
+|                      |False               | -            | |
+|                      |Unknown             | -            | |
+|PIDPressure           |True                | NoSchedule   | node.kubernetes.io/pid-pressure                |
+|                      |False               | -            | |
+|                      |Unknown             | -            | |
+|NodeShutDown          |True                | NoSchedule   | node.cloudprovider.kubernetes.io/shutdown      |
+|                      |False               |-             | |
+|                      |Unknown             |-             | |
+|ExternalCloudProvider |True                | NoSchedule   | node.cloudprovider.kubernetes.io/uninitialized |
```
Comment (bsalamat) [inline contributors/design-proposals/scheduling/taint-node-by-condition.md]: AFAIK, the `ExternalCloudProvider` is not really a node condition. Kubelet taints a node with this taint when the cloud provider is set to external. So, I guess we should explain here that the node is tainted when the cloud provider is external and leave the `Condition Status` column empty.

Comment (mujtabaahmed1991) [reply contributors/design-proposals/scheduling/taint-node-by-condition.md]: @bsalamat I have removed condition column from table, also I have added little explanation about taint when it is set.

File: contributors/design-proposals/scheduling/taint-node-by-condition.md [hunk 2]
```diff
@@ -15,26 +15,30 @@ Currently (1.8 and before), the conditions of nodes are updated by the kubelet a
...
+|                      |False               | -            | |
+|                      |Unknown             | -            | |
+|NodeShutDown          |True                | NoSchedule   | node.cloudprovider.kubernetes.io/shutdown      |
+|                      |False               |-             | |
+|                      |Unknown             |-             | |
+|ExternalCloudProvider |-                   | NoSchedule   | node.cloudprovider.kubernetes.io/uninitialized |
+
+For example, if a CNI network is not detected on the node (e.g. a network is unavailable), the Node Controller will taint the node with `node.kubernetes.io/network-unavailable=:NoSchedule`. This will then allow users to add a toleration to their `PodSpec`, ensuring that the pod can be scheduled to this node if necessary. If cloud provider is set external, kubelet will set taint node with key node.cloudprovider.kubernetes.io/uninitialized. If the kubelet did not update the node’s status after a grace period, the Node Controller will only taint the node with `node.kubernetes.io/unreachable`; it will not taint the node with any unknown condition.
```
Comment (k82cn) [inline contributors/design-proposals/scheduling/taint-node-by-condition.md]: Wrap 'node.cloudprovider.kubernetes.io/uninitialized' with ``

> If cloud provider is set external, kubelet will set taint node with key node.cloudprovider.kubernetes.io/uninitialized

It's better to show "which condition happened" trigger this taint.

File: contributors/design-proposals/scheduling/taint-node-by-condition.md [hunk 3]
```diff
@@ -15,26 +15,30 @@ Currently (1.8 and before), the conditions of nodes are updated by the kubelet a
...
+|DiskPressure          |True                | NoSchedule   | node.kubernetes.io/disk-pressure               |
+|                      |False               | -            | |
+|                      |Unknown             | -            | |
+|NetworkUnavailable    |True                | NoSchedule   | node.kubernetes.io/network-unavailable         |
+|                      |False               | -            | |
+|                      |Unknown             | -            | |
+|PIDPressure           |True                | NoSchedule   | node.kubernetes.io/pid-pressure                |
+|                      |False               | -            | |
+|                      |Unknown             | -            | |
+|NodeShutDown          |True                | NoSchedule   | node.cloudprovider.kubernetes.io/shutdown      |
```
Comment (yastij) [inline contributors/design-proposals/scheduling/taint-node-by-condition.md]: @bsalamat - AFAIK we didn't introduce a nodeShutdown, so there is no equivalent to the taint right now.

**Pull request closed without merging**